sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2AlterTableCommands.scala

```diff
@@ -228,6 +228,15 @@ case class AlterColumn(
       TableChange.updateColumnPosition(colName, newPosition.position)
     }
     val defaultValueChange = setDefaultExpression.map { newDefaultExpression =>
+      if (newDefaultExpression.nonEmpty) {
```

Comment:
If I read the javadoc correctly, I think the new default expression can't be an empty string
```
    /**
     * Returns the column default value SQL string (Spark SQL dialect). The default value literal
     * is not provided as updating column default values does not need to back-fill existing data.
     * Null means dropping the column default value.
     */
    @Nullable
    public String newDefaultValue() { return newDefaultValue; }
```

Reply:
The new default expression can't be an empty string because that will not parse as a SQL expression. Therefore it is acceptable to use the empty string as a sentinel value to indicate that no default value is associated with the target column.

Reply:
Maybe just put a comment here saying "We call 'ResolveDefaultColumns.analyze' here to make sure that the default value parses successfully, and return an error otherwise"?

Reply:
ah, so the javadoc in `TableChange.java` is wrong. @Hisoka-X can you help to fix it?